userspace/libpman/src/configuration.c

```diff
@@ -165,15 +169,126 @@ int pman_get_required_buffers()
 	return g_state.n_required_buffers;
 }
 
+static char byte_array[] = "BPF_TRACE_RAW_TP";
+
+bool check_BPF_TRACE_RAW_TP(void)
+{
+	// These locations are taken from libbpf library:
+	// https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/btf.c#L4767
+	const char *locations[] = {
+		/* try canonical vmlinux BTF through sysfs first */
+		"/sys/kernel/btf/vmlinux",
```

Comment:
Do we have access to `scap_get_host_root` from pman? It'd be great if this didn't require precision mounts from a user.

Reply:
Uhm I need to check it. The point is that even if here we use HOST_ROOT under the hood libbpf does exactly this same check with these paths to find btf on the system... So in this function we will search under /HOST_ROOT/boot/... while in libbpf case we will search under /boot/... Probably is better to use the same path between our code and libbpf, WDYT?

Reply:
`/sys/kernel/btf/vmlinux` should succeed in most cases also the container one as you highlighted Andrea. But am gonna +1 @Molter73 comment here. We know that when deploying in a container the other paths definitely won't exist. So I would add `scap_get_host_root` to be consistent with scap as well. That way we would check valid paths if we even get to them in the loop.

See also our guidance on host mounts needed https://falco.org/docs/install-operate/running/#docker-least-privileged

Reply:
I got your point, what is unclear to me is how `scap_get_host_root` could solve our issues here... 
* Let's say I add `scap_get_host_root` to this method and users will run a container with `-v /boot:/host/boot` so with `HOST_ROOT=/host`. 
* Imagine the case in which the vmlinux is not found under `/sys/kernel/btf/vmlinux` we will search it under  `/host/boot/vmlinux-%1$s` and we will find it.
* Now our check passes so we go on and we try to load the modern bpf with libbpf and we search for the btf file here https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/btf.c#L4773.
*  we won't fine the vmlinux under `/sys/kernel/btf/vmlinux` as before, we will search it under `/boot/vmlinux-%1$s` but we won't find anything since `/boot` is mounted under `/host/boot`

Reply:
I'm a bit torn on this one, I guess since `libbpf` limits us then we'll just have to keep it as is, `/sys/kernel/btf/vmlinux` should be enough on most cases anyways so, meh. WDYT @incertum?

Reply:
Yeah can't have nice things here @Molter73! As everyone is aware `/sys/kernel/btf/vmlinux` should be valid in probably almost all cases, so maybe we are over discussing something not relevant here. Let's merge then!

Reply:
Agree let's keep the paths as they are in libbpf